client/vaultclient/vaultclient_test.go

```diff
@@ -0,0 +1,259 @@
+  capabilities = ["read"]
+}
+
+path "secret/metadata/{{identity.entity.aliases.<<.JWTAuthAccessorID>>.metadata.nomad_namespace}}/*" {
+  capabilities = ["list"]
+}
+
+path "secret/metadata/*" {
+  capabilities = ["list"]
+}
+`
+
+	widVaultRole = `
+{
+  "role_type": "jwt",
+  "bound_audiences": "vault.io",
+  "user_claim": "/nomad_job_id",
+  "user_claim_json_pointer": true,
+  "claim_mappings": {
+    "nomad_namespace": "nomad_namespace",
+    "nomad_job_id": "nomad_job_id"
+  },
+  "token_ttl": "30m",
+  "token_type": "service",
+  "token_period": "72h",
+  "token_policies": ["nomad-workloads"]
+}
+`
+)
+
+func renderVaultTemplate(tmplStr string, data any) ([]byte, error) {
+	var buf bytes.Buffer
+	tmpl, err := template.New("policy").
+		Delims("<<", ">>").
+		Parse(tmplStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse policy template: %w", err)
+	}
+
+	err = tmpl.Execute(&buf, data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to render policy template: %w", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+func setupVaultForWorkloadIdentity(v *testutil.TestVault, jwksURL string) error {
+	logical := v.Client.Logical()
+	sys := v.Client.Sys()
+	ctx := context.Background()
+
+	// Enable JWT auth method.
+	err := sys.EnableAuthWithOptions(jwtAuthMountPathTest, &api.MountInput{
+		Type: "jwt",
+	})
+	if err != nil {
+		return fmt.Errorf("failed to enable JWT auth method: %w", err)
+	}
+
+	secret, err := logical.Read(fmt.Sprintf("sys/auth/%s", jwtAuthMountPathTest))
+	jwtAuthAccessor := secret.Data["accessor"].(string)
+
+	// Write JWT auth method config.
+	jwtAuthConfigData := struct {
+		JWKSURL string
+	}{
+		JWKSURL: jwksURL,
+	}
+	jwtAuthConfig, err := renderVaultTemplate(jwtAuthConfigTemplate, jwtAuthConfigData)
+	if err != nil {
+		return err
+	}
+
+	_, err = logical.WriteBytesWithContext(ctx, fmt.Sprintf("auth/%s/config", jwtAuthMountPathTest), jwtAuthConfig)
+	if err != nil {
+		return fmt.Errorf("failed to write JWT auth method config: %w", err)
+	}
+
+	// Write Nomad workload policy.
+	data := struct {
+		JWTAuthAccessorID string
+	}{
+		JWTAuthAccessorID: jwtAuthAccessor,
+	}
+	policy, err := renderVaultTemplate(widVaultPolicyTemplate, data)
+	if err != nil {
+		return err
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(policy)
+	policyReqBody := fmt.Sprintf(`{"policy": "%s"}`, encoded)
+
+	policyPath := "sys/policies/acl/nomad-workloads"
+	_, err = logical.WriteBytesWithContext(ctx, policyPath, []byte(policyReqBody))
+	if err != nil {
+		return fmt.Errorf("failed to write policy: %w", err)
+	}
+
+	// Write Nomad workload role.
+	rolePath := fmt.Sprintf("auth/%s/role/nomad-workloads", jwtAuthMountPathTest)
+	_, err = logical.WriteBytesWithContext(ctx, rolePath, []byte(widVaultRole))
+	if err != nil {
+		return fmt.Errorf("failed to write role: %w", err)
+	}
+
+	return nil
+}
+
+func TestVaultClient_DeriveTokenWithJWT(t *testing.T) {
```

Comment:
Do we have a test helper that can skip the test if there's no Vault binary around? That might be nice to reduce wait times and spurious errors for non-CI development.

Reply:
Good point, I will poke around other tests like this 👍

Reply:
I didn't find any check like this so I updated `NewTestVault` in https://github.com/hashicorp/nomad/pull/18534/commits/be3cedd89c52a7083315dd14959769db7fd124a4 to skip the test if the binary is not found.